src/configure/configure.ts

```diff
@@ -501,22 +501,24 @@ class PipelineConfigurer {
         }
 
         try {
-            let commitOrDiscard = await vscode.window.showInformationMessage(utils.format(Messages.modifyAndCommitFile, Messages.commitAndPush, this.inputs.sourceRepository.branch, this.inputs.sourceRepository.remoteName), Messages.commitAndPush, Messages.discardPipeline);
-            if (commitOrDiscard && commitOrDiscard.toLowerCase() === Messages.commitAndPush.toLowerCase()) {
-                await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.configuringPipelineAndDeployment }, async (progress) => {
-                    try {
-                        // handle when the branch is not upto date with remote branch and push fails
-                        this.inputs.sourceRepository.commitId = await this.localGitRepoHelper.commitAndPushPipelineFile(this.inputs.pipelineParameters.pipelineFilePath, this.inputs.sourceRepository);
-                    }
-                    catch (error) {
-                        telemetryHelper.logError(Layer, TracePoints.CheckInPipelineFailure, error);
-                        throw (error);
-                    }
-                });
-            }
-            else {
-                telemetryHelper.setTelemetry(TelemetryKeys.PipelineDiscarded, 'true');
-                throw new UserCancelledError(Messages.operationCancelled);
+            while (!this.inputs.sourceRepository.commitId) {
+                let commitOrDiscard = await vscode.window.showInformationMessage(utils.format(Messages.modifyAndCommitFile, Messages.commitAndPush, this.inputs.sourceRepository.branch, this.inputs.sourceRepository.remoteName), Messages.commitAndPush, Messages.discardPipeline);
+                if (commitOrDiscard && commitOrDiscard.toLowerCase() === Messages.commitAndPush.toLowerCase()) {
+                    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.configuringPipelineAndDeployment }, async (progress) => {
+                        try {
+                            // handle when the branch is not upto date with remote branch and push fails
+                            this.inputs.sourceRepository.commitId = await this.localGitRepoHelper.commitAndPushPipelineFile(this.inputs.pipelineParameters.pipelineFilePath, this.inputs.sourceRepository);
+                        }
+                        catch (error) {
+                            telemetryHelper.logError(Layer, TracePoints.CheckInPipelineFailure, error);
+                            vscode.window.showErrorMessage(utils.format(Messages.commitFailedErrorMessage, error.message));
```

Comment:
You are not throwing the error here after catching. I think even though we show error message here to the user, the execution of code will continue and pipeline will be configured. Am I missing anything here?